src/sentry/taskworker/registry.py

```diff
@@ -126,13 +126,28 @@ def wrapped(func: Callable[P, R]) -> Task[P, R]:
         return wrapped
 
     def send_task(self, activation: TaskActivation, wait_for_delivery: bool = False) -> None:
-        metrics.incr("taskworker.registry.send_task", tags={"namespace": activation.namespace})
+        metrics.incr(
+            "taskworker.registry.send_task",
+            tags={"namespace": activation.namespace},
+        )
 
         topic = self.router.route_namespace(self.name)
         produce_future = self._producer(topic).produce(
             ArroyoTopic(name=topic.value),
             KafkaPayload(key=None, value=activation.SerializeToString(), headers=[]),
         )
+        try:
+            if produce_future.exception():
```

Comment:
Should we have a timeout for how long we're willing to wait for an exception? Currently this looks like it would block until we hear back from the broker.

Reply:
Good catch, added a 1 second timeout. Though, this would mean the metric wouldn't capture futures that error after 1 second.